analyses/fusion_filtering/04-project-specific-filtering.Rmd

```diff
@@ -0,0 +1,269 @@
+```{r}
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+setwd(root_dir)
+#load required packages
+suppressPackageStartupMessages(library("readr"))
+suppressPackageStartupMessages(library("tidyverse"))
+suppressPackageStartupMessages(library("reshape2"))
+suppressPackageStartupMessages(library("qdapRegex"))
+
+#read filtFusion files
+strandedQCGeneFiltered_filtFusion<-readRDS(params$dataStranded)
+polyaQCGeneFiltered_filtFusion<-readRDS(params$dataPolya)
+
+QCGeneFiltered_filtFusion<-rbind(strandedQCGeneFiltered_filtFusion,polyaQCGeneFiltered_filtFusion)
+
+write.table(QCGeneFiltered_filtFusion,"analyses/fusion_filtering/results/FilteredFusion.tsv",sep="\t",quote=FALSE,row.names = FALSE)
+
+# subset for recurrent fusion detection and multifused genes QC
+fusion_calls<-unique(QCGeneFiltered_filtFusion)
+# remove distance from intergenic fusions
+fusion_calls$FusionName<-unlist(lapply(fusion_calls$FusionName,function(x) rm_between(x, "(", ")", extract = F)))
+
+
+# get grouping column id
+group<-params$group
+
+# get histology file
+clinical<-read.delim(params$histology,stringsAsFactors = FALSE)
+
+# Least number of callers
+numCaller<-params$numCaller
+
+# Least number of samples per group
+numSample<-params$numSample
+
+# Max number of groups 
+numGroup<-params$numGroup
+
+# Max number of times gene can be fused per sample
+limitMultiFused<-params$limitMultiFused
+
+print("Raw calls from STARfusion and Arriba for PBTA")
+table(fusion_calls$Caller)
+```
+
+
+```{r}
+
+# aggregate caller 
+fusion_caller.summary <- fusion_calls %>%
+  dplyr::filter(Fusion_Type != "other") %>%
+  dplyr::select(Sample,FusionName,Caller,Fusion_Type) %>%
+  group_by(FusionName, Sample ,Fusion_Type) %>%
+  unique() %>%
+  dplyr::mutate(CalledBy = toString(Caller), caller.count = n()) %>%
+  dplyr::select(-Caller)
+  
+# # aggregate by read count
+#  fusion_read.summary <- fusion_calls %>%
+#    dplyr::filter(Fusion_Type != "other") %>%
+#    dplyr::select(Sample,FusionName,JunctionReadCount,SpanningFragCount) %>%
+#    group_by(FusionName, Sample ) %>%
+#    unique() %>%
+#    dplyr::mutate(JunctionReadCountSum=sum(JunctionReadCount),SpanningFragCountSum=sum(SpanningFragCount))
+# 
+# fusion_summary<-left_join(fusion_caller.summary,fusion_read.summary,by=c("Sample","FusionName"))
+#  
+
+# remove fusion within local rearrangement
+fusion_calls <- fusion_calls %>%
+  # remove local rearrangement/adjacent genes
+  dplyr::filter(!grepl("LOCAL_REARRANGEMENT|LOCAL_INVERSION",annots))
+
+#to add aggregated caller from fusion_caller.summary
+fusion_calls<-fusion_calls %>% 
+  dplyr::filter(Fusion_Type != "other") %>% dplyr::select(-Caller,-annots) %>%
+  left_join(fusion_caller.summary,by=(c("Sample","FusionName","Fusion_Type"))) %>%
+  dplyr::select(-JunctionReadCount,-SpanningFragCount,-Confidence,-LeftBreakpoint,-RightBreakpoint) %>% unique()
+
+#merge with histology file
+fusion_calls<-merge(fusion_calls,clinical,by.x="Sample",by.y="Kids_First_Biospecimen_ID")
```

Comment:
Are you using anything besides `broad_histology` from this file? I am wondering for a few reasons -- it's memory efficient to get rid of the columns you don't need and I noticed in `results/PutativeDriverFusion.tsv` that that contains lots of the clinical metadata. 

I can also imagine a situation where someone is working with this version of the fusion files and then doesn't need to join with the histologies file. That could be convenient, but the histologies file might be updated in a later release and this could be rendered out of date. I think you want to design the output of this keeping in mind that things are consistently updated.

Reply:
@kgaonkar6 and @jaclyn-taroni I suggest outputting only PT_ID and BS_ID as metadata with these results so it can be merged with future clinical later, as @jaclyn-taroni  suggested

Reply:
Im only keeping Kids_First_Biospecimen_ID, Kids_First_Participant_ID and broad_histology now. The broad_histology is used to check for n samples per group etc. Should that be ok?

Reply:
In this case, it is not necessary since we are only producing a table with the final fusions - only useful for plotting. We are just talking about the final TSV for use in oncoprints, correct? As a supplemental table, we probably would want to add that information. It is possible `broad_histology` would change if we find samples mis-diagnosed...

Reply:
I am inclined to leave `broad_histology` out of the final table as well. The "final" TSV may change if samples have `broad_histology` labels change, but this is all set up to be easily rerun!